README.md

```diff
@@ -11,8 +11,69 @@ it to 'deploy' directory of target nexus repository manager. For more informatio
 "[Installing a custom Nexus 3 plugin](https://sonatype-nexus-community.github.io/nexus-development-guides/plugin-install.html)".
 
 # Usage
-This plugin provides tagging functionality by exposing RESTFul API. See swagger documentation in System->API section of 
-Administration panel to see how this can be used.
+This plugin provides tagging functionality by exposing RESTFul API. Each tag is a document with a unique name and a
+set of optional attributes. Each tags can be associated with one or more components (artifacts).  Stored tags can be
+searched via RESTFul API. Search API supports searching tags by attributes or associated components. APIs can be called
+manually in 'System->API' section of Administration panel.
+
+Here are few examples of using RESTFul APIs:
+
+Adding a tag:
+```
+curl -X POST --header 'Content-Type: application/json' http://127.0.0.1:8081/service/rest/v1/tags \
+  -d '{
+    "name": "project1-142",
+    "attributes": {
+        "version": "1",
+        "status": "failed"
+    },
+    "components" : [
+        {
+          "repository": "repo1",
+          "group": "gr1",
+          "name": "comp1",
+          "version": "1"
+        }
+    ]
+}' 
+```
+
+Updating a tag:
+```
+curl -X PUT --header 'Content-Type: application/json' http://127.0.0.1:8081/service/rest/v1/tags/project1-142 \
+  -d '{
+    "name": "project1-142",
+    "attributes": {
+        "version": "1",
+        "status": "successful"
+    },
+    "components" : [
+        {
+          "repository": "repo1",
+          "group": "gr1",
+          "name": "comp1",
+          "version": "1"
+        },
+        {
+          "repository": "repo1",
+          "group": "gr1",
+          "name": "comp2",
+          "version": "3"
+        }
+    ]
+}' 
+```
+
+Searching tags:
+```
+curl -X GET --header "accept: application/json" \
```

Comment:
please explain search parameters and it's capabilities.

Reply:
Done